consul/src/test/java/com/linecorp/armeria/client/consul/ConsulEndpointGroupTest.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.consul;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.common.SessionProtocol;
+import com.linecorp.armeria.internal.consul.ConsulTestBase;
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.ServerListener;
+import com.linecorp.armeria.server.consul.ConsulUpdatingListener;
+
+public class ConsulEndpointGroupTest extends ConsulTestBase {
+
+    static final List<Server> servers = new ArrayList<>();
+
+    @BeforeAll
+    static void startServers() {
+
+        for (Endpoint endpoint : sampleEndpoints) {
+            final Server server = Server.builder()
+                                        .http(endpoint.port())
+                                        .service("/", new EchoService())
+                                        .build();
+            final ServerListener listener = ConsulUpdatingListener.builder(serviceName)
+                                                                  .consulProtocol(SessionProtocol.HTTP)
+                                                                  .consulAddress(consul().getAddress())
+                                                                  .consulPort(consul().getHttpPort())
+                                                                  .consulApiVersion("v1")
```

Comment:
Programatically build looks good. However, some users might want to load a Consul URI from their configuration files such as `yml` or `properties` and create this builder the `URI` string. 
Don't we need to offer `consulUri(String)` and `consulUri(URI)`? 🤔

Reply:
Maybe there is a use case for that. If so I think the API version string is not included in the URI. What is your opinion?

Reply:
How about mentioning the Consul URI spec that should contain an API version in the Javadoc?

Reply:
I did. PTAL.